### PR TITLE
Add Script File feature

### PR DESCRIPTION
**Pull request by nicolassing**
Hello, 

I've added scripting file-base feature in addition of inline scripting.

### CODE REVIEW

File: lib/Elastica/AbstractScript.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * @author Nicolas Assing <nicolas.assing@gmail.com>
+ *
+ * @link http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting.html
+ */
+abstract class AbstractScript extends AbstractUpdateAction
+{
+    /**
+     * @param array|null $params
+     * @param null $id
```
Comment (im-denisenko) [inline lib/Elastica/AbstractScript.php]: `@param string $id`

**Comment by im-denisenko** [comment]
@nicolassing Looks good, thanks.
Build failed, but this is not related to PR.
Could you please add a few functional tests and run cs fixer using `make lint` before merge?

**Comment by ruflin** [comment]
@nicolassing Any updates here? I plan to do a release today and would like to include this. Make sure merge master as quite a few changes went into master.

**Comment by nicolassing** [comment]
@ruflin I've added a functional test. But this functional test will not work without the script file. I've updated elasticsearch configuration and created a script file (calculate-distance.groovy) in test/data/ but this script file must be in /etc/elasticsearch/scripts. I don't know how to do this with Docker and Vagrant :(

**Comment by ruflin** [comment]
Thx, I will have a look.

**Comment by ruflin** [comment]
@nicolassing After fighting a  little bit with ansible I got it running here: https://github.com/ruflin/Elastica/pull/914 I had to move the files from the data to the config/scripts directory as elasticsearch loads the files from there.

**Comment by ruflin** [comment]
I merged my pull request with the changes. Thanks for the contribution.

**Comment by nicolassing** [comment]
:+1: Someday i'd learn how to really use ansible, vagrant, docker, etc ... Someday ... :smile:

**Comment by ruflin** [comment]
You can directly jump to docker ;-)

**Pull request merged**